GitUI/CommandsDialogs/FormDeleteBranch.cs

```diff
@@ -48,15 +48,22 @@ protected override void OnRuntimeLoad(EventArgs e)
             base.OnRuntimeLoad(e);
 
             Branches.BranchesToSelect = Module.GetRefs(RefsFilter.Heads).ToList();
-            foreach (string branch in Module.GetMergedBranches())
+            if (AppSettings.DontConfirmDeleteUnmergedBranch)
             {
-                if (!branch.StartsWith("* "))
-                {
-                    _mergedBranches.Add(branch.Trim());
-                }
-                else if (!branch.StartsWith("* ") || !DetachedHeadParser.IsDetachedHead(branch[2..]))
+                _currentBranch = Module.GetSelectedBranch();
```

Comment:
Why is this needed?

Reply:
For the check to not delete the current branch.
In the `else` branch (with the original code), it is taken from the output of `GetMergedBranches()`.